libs/cart/guides/loading-states.md

```diff
@@ -0,0 +1,11 @@
+
+## Cart Item States
+There are times when the app developer not only wants to show that the cart item feature area is loading, but that a particular cart item is loading or has a particular state. For example, one might want to indicate that a recently added cart item was recently added, or that a cart item whose quantity is being changed is mutating. For these use cases daffodil provides states for each cart item. The possible states for a cart item are new, mutating, updated, and default.
+
+## Edge Cases
+While we try to leave as much creative freedom as possible in the hands of the app developer, there are some edge cases on which we needed to make a decision in order for this state system to work. Here are a list of edge cases where it isn't very obvious how daffodil state would work:
+- When a customer updates or deletes a cart item, the cart items feature state will remain in a "completed" state, but the individual cart item state will be go from "default" to "mutating" to "updated". On the other hand, when a customer adds a product to the cart, the cart items feature state will go into a mutating state, but the cart item state will go from "default" directly to a "new" state. This is to enable the app developer to show loading states for particular cart items that are being mutated or deleted, while avoiding the complexity that comes with trying to show individual loading states for cart items being added to the cart. Some parts of this complexity are questions like: How do I indicate to the customer that a new cart item is being added to the cart if that cart item does not exist in state until after the item has been added? Possible to solve, but ugly. How do I discern if a composite product being added already exists on the cart or if it will be a new cart item? The difficulty here is that the variantId is needed to add a composite product to the cart to distinguish it from other configurations of the same composite product, but this variantId isn't returned on the DaffCompositeCartItem. The only solves to this problem are either to require an additional field (an array of the selected options) in the DaffCartItemAdd action, or to change the backend to return the variantId associated with a composite cart item. Neither of these are particularly attractive solutions, so we elected to simply cause the entire Cart Item Feature area to load whenever a product is added to the cart.
```

Comment:
```
Some parts of this complexity are questions like: How do I indicate to the customer that a new cart item is being added to the cart if that cart item does not exist in state until after the item has been added? Possible to solve, but ugly. How do I discern if a composite product being added already exists on the cart or if it will be a new cart item?
```

Wrap these questions with quotations: `"`.

Reply:
done